File: Zamba/ZambaWeb2013/zamba2010/Zamba.WFActivity.Regular/Action/DoGetNewId.vb
Imports Zamba.Data
Imports Zamba.Core
''Imports Zamba.WFBusiness
Imports System.Xml.Serialization
''' <summary>
'''  Genera un Id y completa un indice determinado.
''' </summary>
''' <remarks></remarks>
<RuleCategory("Datos"), RuleDescription("Generar Id"), RuleHelp("Genera un nuevo id e ingresa el nuevo valor del indice en el entidad seleccionado"), RuleFeatures(False)> <Serializable()> _
Public Class DoGetNewId
    Inherits WFRuleParent
    Implements IDoGetNewId
    Private _isLoaded As Boolean
    Private _isFull As Boolean
    Public Overrides ReadOnly Property IsFull() As Boolean
        Get
            Return _isFull
        End Get
    End Property
    Public Overrides ReadOnly Property IsLoaded() As Boolean
        Get
            Return _isLoaded
        End Get
    End Property
    Public Overrides Sub FullLoad()

    End Sub
    Public Overrides Sub Load()

    End Sub
    Private m_iDocTypeId As Int32
    Private m_iIndexId As Int32
    Public Overrides Sub Dispose()

    End Sub
#Region "Constructor"
    Public Sub New(ByVal Id As Int64, ByVal Name As String, ByVal wfstepid As Int64, ByVal p_iDocTypeId As Int32, ByVal p_iIndexId As Int32)
        MyBase.New(Id, Name, wfstepid)
        m_iDocTypeId = p_iDocTypeId
        m_iIndexId = p_iIndexId
    End Sub
#End Region

    Public Property DocTypeId() As Int32 Implements IDoGetNewId.DocTypeId
        Get
            Return m_iDocTypeId
        End Get
        Set(ByVal value As Int32)
            m_iDocTypeId = value
        End Set
    End Property

    Public Property IndexId() As Int32 Implements IDoGetNewId.IndexId
        Get
            Return m_iIndexId
        End Get
        Set(ByVal value As Int32)
            m_iIndexId = value
        End Set
    End Property

    '[Ezequiel] 09/06/2009 - Se elimino el Try-Catch
    Public Overloads Overrides Function Play(ByVal results As System.Collections.Generic.List(Of ITaskResult)) As System.Collections.Generic.List(Of ITaskResult)
        Dim playRule As New Zamba.WFExecution.PlayDoGetNewId()
        Return playRule.Play(results, Me)
    End Function
    Public Overloads Overrides Function Playweb(ByVal results As System.Collections.Generic.List(Of ITaskResult), ByRef RulePendingEvent As RulePendingEvents, ByRef ExecutionResult As RuleExecutionResult, ByRef Params As Hashtable) As System.Collections.Generic.List(Of ITaskResult)
        Dim playRule As New Zamba.WFExecution.PlayDoGetNewId()
        Return playRule.Play(results, Me)
    End Function
End Class

File: Zamba/ZambaWeb2013/zamba2010/Zamba.WorkFlow.SQLExecution/Zamba.WorkFlow.Execution.ActivityLibrary/Activity/DoGetNewId.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Reflection;
using System.Diagnostics;
using Zamba.Core;
//using Zamba.WFUserControl;

namespace Zamba.WFActivity.Xoml
{
    /// <summary>
    /// This activity show a ArrayList of delays
    /// </summary>
    [RuleDescription("GENERAR ID"), RuleHelp("Genera un id y completa un indice determinado.")]
    public partial class DoGetNewId : ZRule, IResultActivity, IDoGetNewId
	{
        public DoGetNewId()
		{
			InitializeComponent();
		}

        protected override ActivityExecutionStatus Execute(ActivityExecutionContext executionContext)
        {
            Zamba.WFExecution.PlayDoGetNewId play =
                new Zamba.WFExecution.PlayDoGetNewId();
            play.Play(Results, this);
            return ActivityExecutionStatus.Closed;
        }

        Int32 m_iDocTypeId;
        Int32 m_iIndexId;

        public override void SetParams(object[] Params)
        {
            this.m_iDocTypeId = Int32.Parse(Params[0].ToString());
            this.m_iIndexId = Int32.Parse(Params[1].ToString());
        }

        public override System.Collections.Generic.List<object> GetParams()
        {
            System.Collections.Generic.List<object> Params = new System.Collections.Generic.List<object>();
            Params.Add(this.m_iDocTypeId);
            Params.Add(this.m_iIndexId);
            return Params;
        }

        #region Miembros de IDoGetNewId

        public int DocTypeId
        {
            get
            {
                return m_iDocTypeId;
            }
            set
            {
                m_iDocTypeId = value;
            }
        }

        public int IndexId
        {
            get
            {
                return m_iIndexId;
            }
            set
            {
                m_iIndexId = value;
            }
        }

        #endregion
    }
}
